pages/contentful/slug-validation/index.vue

```diff
@@ -0,0 +1,202 @@
+        slugField: null,
+        titleField: null,
+        slugFromTitle: false,
+        errorMessage: null,
+        debouncedDuplicateStatus: null,
+        contentfulExtensionSdk: null
+      };
+    },
+
+    computed: {
+      duplicateStatus() {
+        return this.debouncedDuplicateStatus;
+      }
+    },
+
+    watch: {
+      value: 'updateContentfulField',
+      debouncedDuplicateStatus: 'handleStatus'
+    },
+
+    mounted() {
+      window.contentfulExtension.init(sdk => {
+        this.contentfulExtensionSdk = sdk;
+        if (sdk.location.is(window.contentfulExtension.locations.LOCATION_ENTRY_FIELD)) {
+          sdk.window.startAutoResizer();
+
+          this.value = sdk.field.getValue();
+          this.slugField = sdk.field;
+          const titleFieldName = sdk.contentType.displayField;
+          this.titleField = sdk.entry.fields[titleFieldName];
+          this.titleField.onValueChanged(this.handleTitleChange);
+          const titleValue = this.titleField.getValue();
+          if (!this.value || this.value === 'undefined' || this.value === '') {
+            // No slug exists, use the title as a default
+            this.slugFromTitle = true;
+            this.handleTitleChange(titleValue);
+          } else if (this.value === this.convertToSlug(titleValue)) {
+            // slug is the same as title, enable automatic update when the title changes.
+            this.slugFromTitle = true;
+          }
+        }
+      });
+    },
+
+    methods: {
+      updateContentfulField() {
+        if (this.contentfulExtensionSdk) this.contentfulExtensionSdk.field.setValue(this.value);
+      },
+
+      // Convert using the speakingurl package, which is included in the contentful layout.
+      convertToSlug(text) {
+        return window.getSlug(text);
+      },
+
+      /**
+       * Handle change of slug value caused by either changing slug field
+       * or changing the title of the entry
+       */
+      handleValueChange(value) {
+        this.value = value;
+        this.getDebouncedDuplicateStatus(value);
+      },
+
+      /**
+       * Handle change of slug value caused by changing the slug of the entry.
+       * This will disable updates of the slug from the title for the  current form.
+       */
+      handleSlugChange(value) {
+        this.slugFromTitle = false;
+        this.handleValueChange(value);
+      },
+
+      /**
+       * Handle change of title value caused by changing the title of the entry.
+       * If the slug was changed independently, don't update based of title changes.
+       */
+      handleTitleChange(value) {
+        if (this.slugFromTitle) this.handleValueChange(this.convertToSlug(value || ''));
+      },
+
+      /**
+       * Handle validation status, setting the field to valid/invalid and
+       * removing the value in order to prevent publication if the slug is a duplicate.
+       */
+      async handleStatus() {
+        const status = await this.duplicateStatus;
```

Comment:
because the duplicateStatus value would effectively end up as a promise coming from the getDuplicates assignement to debouncedDuplicateStatus.

I have however removed the duplicateStatus computed property as it did  nothing in the end except introduce an extra layer of variables.